### PR TITLE
add prerequisites to tutorials

### PR DESCRIPTION
**Pull request by nswamy**
@zackchase @madjam @piiswrong 
Please review

I have added prerequisites to the main Tutorials page and also to the data iterator tutorial. I feel that these are common and should be at the top of the Tutorials page instead of having them in each tutorial. I will remove one of them based on the feedback I get.

screen shot 2017-06-06 at 8 23 36 pm
screen shot 2017-06-06 at 8 23 13 pm

### CODE REVIEW

File: docs/tutorials/basic/data.md [hunk 1]
```diff
@@ -4,6 +4,46 @@ Most training and inference modules in MXNet accept data iterators,
 which simplifies this procedure, especially when reading large datasets.
 Here we discuss the API conventions and several provided iterators.
 
+## Prerequisites
+
+To complete this tutorial, we need:  
+
+- [Python](https://www.python.org/downloads/)
```
Comment (piiswrong) [inline docs/tutorials/basic/data.md]: Installing mxnet requires python and pip. These are unnecessary

File: docs/tutorials/basic/data.md [hunk 2]
```diff
@@ -4,6 +4,46 @@ Most training and inference modules in MXNet accept data iterators,
...
 
+## Prerequisites
+
+To complete this tutorial, we need:  
+
+- [Python](https://www.python.org/downloads/)  
+
+- [PIP](https://pypi.python.org/pypi)  
+
+- [MXNet](http://mxnet.io/). You can either [build from source](http://mxnet.io/get_started/install.html) or install from [pip](https://pypi.python.org/pypi/mxnet).
```
Comment (piiswrong) [inline docs/tutorials/basic/data.md]: Simply point to mxnet installation guide page

File: docs/tutorials/basic/data.md [hunk 3]
```diff
@@ -4,6 +4,46 @@ Most training and inference modules in MXNet accept data iterators,
...
+
+```bash
+$ pip install requests
+```
+- [Matplotlib](https://matplotlib.org/)  
+
+```bash
+$ pip install matplotlib
+```
+- [Jupyter Notebook](http://jupyter.org/index.html)
```
Comment (piiswrong) [inline docs/tutorials/basic/data.md]: condense all of above to one line
`pip install opencv-python requests ...`

Comment (madjam) [inline docs/tutorials/basic/data.md]: Why repeat this here? Why can't we just point to installation page?

Comment (nswamy) [inline docs/tutorials/basic/data.md]: Installation page does not list all the dependencies(typically any dependency used within MXNet should either come from the PIP package or be a part of installation steps/script). some of the dependencies are only needed for the tutorial.

**Comment by nswamy** [comment]
moved to 
https://github.com/dmlc/mxnet/pull/6639

**Pull request closed without merging**